devtools/etdump/etdump_filter.cpp

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright (c) Meta Platforms, Inc. and affiliates.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+#include <executorch/devtools/etdump/etdump_filter.h>
+
+#include <executorch/runtime/core/error.h>
+
+using ::executorch::runtime::DelegateDebugIntId;
+using ::executorch::runtime::Error;
+using ::executorch::runtime::kUnsetDelegateDebugIntId;
+
+namespace executorch {
+namespace etdump {
+
+ETDumpFilter::ETDumpFilter()
+    : regex_count_(0), range_start_(0), range_end_(0) {}
+
+Result<bool> ETDumpFilter::add_regex(const char* pattern) {
```

Comment:
I think let's call this add_filter, we don't necessarily want to tie this to regex i think. On smaller platforms where they pass their own filter implementation they can just do simple string matching.